tools/release.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import re
+import sys
+
+from utils import (
+    run,
+    repository_dirty,
+    remote_exists,
+    matching_remote_branches,
+    local_tag_exists,
+    remote_tag_exists,
+    local_branch_matches_remote_branch,
+    configure_git_user,
+    create_local_tag,
+    push_tag,
+)
+
+log = logging.getLogger(__name__)
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%SZ",
+)
+
+SEMVER_MAJOR_MINOR_PATTERN = "(0|[1-9]\d*)\.(0|[1-9]\d*)"
+
+# https://regex101.com/r/vkijKf/1/
+SEMVER_PATTERN = "(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
+
+RELEASE_TAG_PATTERN = f"v{SEMVER_PATTERN}-{SEMVER_PATTERN}"
+
+STABLE_BRANCH_NAME_PATTERN = f"release-v{SEMVER_MAJOR_MINOR_PATTERN}"
+
+
+def valid_release_tag(tag: str) -> bool:
+    return bool(re.match(RELEASE_TAG_PATTERN, tag))
+
+
+def valid_stable_branch_name(branch: str) -> bool:
+    return bool(re.match(STABLE_BRANCH_NAME_PATTERN, branch))
+
+
+def validate_arguments_and_environment(
+    git_remote: str, git_branch: str, git_tag: str, debug: bool
+) -> int:
+    if repository_dirty(debug):
+        log.info(
```

Comment:
Shouldn't all these be `log.error` instead?

Reply:
f526b51196f28ac992c08b2a251d3f4ee4da20d9